types.go

```diff
@@ -42,47 +52,52 @@ var (
 //      T *Thingy `optional:"true"`
 //    }
 //
-type In struct{}
+type In struct{ digSentinel }
```

Comment:
I figured since `dig.In` are like sentinel nodes in the graph of embedded types, that terminology could be used.


https://en.wikipedia.org/wiki/Sentinel_node